scipy/stats/_censored_data.py

```diff
@@ -0,0 +1,367 @@
+            x = []
+        if left is None:
+            left = []
+        if right is None:
+            right = []
+        if intervals is None:
+            intervals = np.empty((0, 2))
+
+        _validate_1d(x, 'x')
+        _validate_1d(left, 'left')
+        _validate_1d(right, 'right')
+        x2, left2, right2, intervals2 = _validate_intervals(intervals)
+
+        self._x = np.concatenate((x, x2))
+        self._left = np.concatenate((left, left2))
+        self._right = np.concatenate((right, right2))
+        # Note that by construction, the private attribute _intervals
+        # will be a 2D array that contains only finite values representing
+        # intervals with nonzero but finite length.
+        self._intervals = intervals2
+
+    def __repr__(self):
+        x_str = " ".join(np.array_repr(self._x).split())
+        left_str = " ".join(np.array_repr(self._left).split())
+        right_str = " ".join(np.array_repr(self._right).split())
+        intervals_str = " ".join(np.array_repr(self._intervals).split())
+        return (f"CensoredData(x={x_str}, left={left_str}, right={right_str},"
+                f" intervals={intervals_str})")
+
+    def __str__(self):
+        num_nc = len(self._x)
+        num_lc = len(self._left)
+        num_rc = len(self._right)
+        num_ic = len(self._intervals)
+        n = num_nc + num_lc + num_rc + num_ic
+        parts = [f'{num_nc} not censored']
+        if num_lc > 0:
+            parts.append(f'{num_lc} left-censored')
+        if num_rc > 0:
+            parts.append(f'{num_rc} right-censored')
+        if num_ic > 0:
+            parts.append(f'{num_ic} interval-censored')
+        return f'CensoredData({n} values: ' + ', '.join(parts) + ')'
+
+    # This is not a complete implementation of the arithmetic operators.
+    # All we need is subtracting a scalar and dividing by a scalar.
+
+    def __sub__(self, other):
+        return CensoredData(x=self._x - other,
+                            left=self._left - other,
+                            right=self._right - other,
+                            intervals=self._intervals - other)
+
+    def __truediv__(self, other):
+        return CensoredData(x=self._x / other,
+                            left=self._left / other,
+                            right=self._right / other,
+                            intervals=self._intervals / other)
+
+    def __len__(self):
+        """
+        The number of values (censored and not censored).
+        """
+        return (len(self._x) + len(self._left) + len(self._right)
+                + len(self._intervals))
+
+    def num_censored(self):
+        """
+        Number of censored values.
+        """
+        return len(self._left) + len(self._right) + len(self._intervals)
+
+    @classmethod
+    def right_censored(cls, x, censored):
+        """
+        Create a `CensoredData` instance of right-censored data.
+
+        Parameters
+        ----------
+        x : array_like
```

Comment:
We should also say 1D as to match the doc of the init.
```suggestion
        x : array_like, 1D
```